packages/@storefront/core/src/core/tag/decorators.ts

```diff
@@ -81,3 +82,33 @@ export function origin(name: string) {
 export function configurable<P extends object>(target: TagConstructor) {
   utils.setMetadata(target, CONFIGURABLE_KEY, true);
 }
+
+export function uiState<P extends object = any, S extends object = any>(
+  prop: string = 'uiValue',
```

Comment:
Do we need this to be overridable? I would stick with one prop name for consistency. You can add it to the tag index file to add it for all tags (I think we do that for `storeSection`). I would also call it `uiKey` or something like that, because the value is being used as a key.